src/renderer/atlas/AtlasEngine.r.cpp

```diff
@@ -234,55 +234,53 @@ void AtlasEngine::_adjustAtlasSize()
     }
     {
         static constexpr D2D1_COLOR_F color{ 1, 1, 1, 1 };
-        wil::com_ptr<ID2D1SolidColorBrush> brush;
-        THROW_IF_FAILED(_r.d2dRenderTarget->CreateSolidColorBrush(&color, nullptr, brush.addressof()));
-        _r.brush = brush.query<ID2D1Brush>();
+        THROW_IF_FAILED(_r.d2dRenderTarget->CreateSolidColorBrush(&color, nullptr, _r.brush.put()));
+        _r.brushColor = 0xffffffff;
     }
 
+    _r.atlasSizeInPixel = requiredSize;
+    _r.atlasBuffer = std::move(atlasBuffer);
+    _r.atlasView = std::move(atlasView);
+    _setShaderResources();
+
     WI_SetAllFlags(_r.invalidations, RenderInvalidations::ConstBuffer);
     WI_SetFlagIf(_r.invalidations, RenderInvalidations::Cursor, !copyFromExisting);
 }
 
 void AtlasEngine::_processGlyphQueue()
 {
-    if (_r.glyphQueue.empty())
+    if (_r.glyphQueue.empty() && WI_IsFlagClear(_r.invalidations, RenderInvalidations::Cursor))
     {
         return;
     }
 
     _r.d2dRenderTarget->BeginDraw();
-    for (const auto& pair : _r.glyphQueue)
+
+    if (WI_IsFlagSet(_r.invalidations, RenderInvalidations::Cursor))

```

Comment:
note: cursor drawing moved here

Reply:
This is so that I avoid calling `BeginDraw()` twice and allow D2D to batch up cursor and glyph drawing in a single render pass.